File: msu_platform/frontend/check_imports.py
import os
import re

hooks = [
    'useState', 'useEffect', 'useRef', 'useCallback', 'useMemo', 
    'useContext', 'useReducer', 'useLayoutEffect',
    'useNavigate', 'useLocation', 'useParams', 'useSearchParams',
    'useQuery', 'useMutation', 'useQueryClient'
]

def check_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    missing = []
    for hook in hooks:
        # Check if hook is used (e.g. useState( or useState<)
        if re.search(r'\b' + hook + r'\b\s*[<(]', content):
            # Check if imported
            # import { useState } from 'react'
            # import { useQuery } from '@tanstack/react-query'
            if not re.search(r'import\s+.*?\b' + hook + r'\b', content):
                if not re.search(r'React\.' + hook, content):
                    missing.append(hook)
                    
    if missing:
        print(f"File: {filepath} missing imports: {', '.join(missing)}")

for root, dirs, files in os.walk('c:/Users/User/Desktop/clubs/clubs/msu_platform/frontend/src'):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            check_file(os.path.join(root, file))

File: msu_platform/frontend/check_imports.js
const fs = require('fs');
const path = require('path');

const hooks = ['useState', 'useEffect', 'useRef', 'useCallback', 'useMemo'];

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else {
            if (file.endsWith('.tsx') || file.endsWith('.ts')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk('c:/Users/User/Desktop/clubs/clubs/msu_platform/frontend/src');

for (const file of files) {
    const content = fs.readFileSync(file, 'utf8');
    
    // Check if any hook is used
    let usedHooks = [];
    for (const hook of hooks) {
        // Find usages like `useState(` or `useState<`
        if (new RegExp(`\\b${hook}\\b\\s*[<(]`).test(content)) {
            usedHooks.push(hook);
        }
    }
    
    if (usedHooks.length > 0) {
        // Check if react imports them
        // import { useState } from 'react';
        // import React, { useState } from 'react';
        let missing = [];
        for (const hook of usedHooks) {
            // Very naive check: does the file contain "hook" and "react"?
            // A better check:
            const importRegex = new RegExp(`import\\s+.*\\b${hook}\\b.*from\\s+['"]react['"]`);
            if (!importRegex.test(content)) {
                // Check if they use React.useState instead
                const reactDotRegex = new RegExp(`React\\.${hook}`);
                if (!reactDotRegex.test(content)) {
                   missing.push(hook);
                }
            }
        }
        
        if (missing.length > 0) {
            console.log(`Missing imports in ${file}: ${missing.join(', ')}`);
        }
    }
}
